src/graphic/Text.ts

```diff
@@ -484,7 +484,12 @@ class ZRText extends Displayable<TextProps> implements GroupLike {
         const needDrawBg = needDrawBackground(style);
         const bgColorDrawn = !!(style.backgroundColor);
 
-        let outerHeight = contentBlock.outerHeight;
+        // outerWidth is the real text width containing padding.
+        const outerHeight = contentBlock.outerHeight;
+        const outerWidth = contentBlock.outerWidth;
+        const innerWidth = textPadding
```

Comment:
Can we provide `contentWidth` like before and avoid calculating it with padding again?